sim/rogue/items_sets_pve.go

```diff
@@ -382,3 +386,46 @@ var ItemSetMadCapsOutfit = core.NewItemSet(core.ItemSet{
 		},
 	},
 })
+
+///////////////////////////////////////////////////////////////////////////
+//                            SoD Phase 6 Item Sets
+///////////////////////////////////////////////////////////////////////////
+
+var ItemSetEmblemsofVeiledShadows = core.NewItemSet(core.ItemSet{
+	Name: "Emblems of Veiled Shadows",
+	Bonuses: map[int32]core.ApplyEffect{
+		// 3 pieces: Your finishing moves cost 50% less Energy.
+		3: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.SpellCode == SpellCode_RogueBetweentheEyes || spell.SpellCode == SpellCode_RogueBladeDance || spell.SpellCode == SpellCode_RogueCrimsonTempest || spell.SpellCode == SpellCode_RogueEnvenom || spell.SpellCode == SpellCode_RogueEviscerate || spell.SpellCode == SpellCode_RogueExposeArmor || spell.SpellCode == SpellCode_RogueRupture || spell.SpellCode == SpellCode_RogueSliceandDice {
+					spell.Cost.Multiplier -= 50
+				}
+			})
```

Comment:
I wonder if we could just create an array of spells on the Rogue struct called `SpenderSpells` where we add all spenders in so that we can apply things like this without the hard-coding?

Reply:
Updated